qiskit/pulse/commands/acquire.py

```diff
@@ -100,54 +101,82 @@ def __repr__(self):
-                       qubits: Union[Qubit, List[Qubit]],
-                       mem_slots: Optional[Union[MemorySlot, List[MemorySlot]]] = None,
-                       reg_slots: Optional[Union[RegisterSlot, List[RegisterSlot]]] = None,
+                       qubit: Union[AcquireChannel, List[AcquireChannel]],
+                       mem_slot: Optional[MemorySlot] = None,
+                       reg_slot: Optional[RegisterSlot] = None,
+                       mem_slots: Optional[List[MemorySlot]] = None,
+                       reg_slots: Optional[List[RegisterSlot]] = None,
                        name: Optional[str] = None) -> 'AcquireInstruction':
-        return AcquireInstruction(self, qubits, mem_slots, reg_slots, name=name)
+
+        return AcquireInstruction(self, qubit, mem_slot, reg_slot,
+                                  mem_slots=mem_slots, reg_slots=reg_slots,
+                                  name=name)
     # pylint: enable=arguments-differ
 
 
 class AcquireInstruction(Instruction):
-    """Pulse to acquire measurement result."""
+    """ Pulse to acquire measurement result. """
 
     def __init__(self,
                  command: Acquire,
-                 acquires: Union[AcquireChannel, List[AcquireChannel]],
-                 mem_slots: Union[MemorySlot, List[MemorySlot]],
-                 reg_slots: Optional[Union[RegisterSlot, List[RegisterSlot]]] = None,
+                 acquire: Union[AcquireChannel, List[AcquireChannel]],
+                 mem_slot: Optional[MemorySlot] = None,
+                 reg_slot: Optional[RegisterSlot] = None,
+                 mem_slots: Optional[List[MemorySlot]] = None,
+                 reg_slots: Optional[List[RegisterSlot]] = None,
                  name: Optional[str] = None):
 
-        if not isinstance(acquires, list):
-            acquires = [acquires]
+        if isinstance(acquire, list) or mem_slots or reg_slots:
+            warnings.warn('AcquireInstruction on multiple qubits, multiple memory slots '
+                          'and multiple reg slots has been deprecated.',
+                          DeprecationWarning, 3)
+
+        if not isinstance(acquire, list):
+            acquire = [acquire]
 
-        if isinstance(acquires[0], Qubit):
+        if isinstance(acquire[0], Qubit):
             raise PulseError("AcquireInstruction can not be instantiated with Qubits, "
                              "which are deprecated.")
 
+        if mem_slot:
+            mem_slots = [mem_slot]
+
+        if reg_slot:
+            reg_slots = [reg_slot]
+
         if not (mem_slots or reg_slots):
             raise PulseError('Neither memoryslots or registers were supplied')
 
-        if mem_slots:
-            if isinstance(mem_slots, MemorySlot):
-                mem_slots = [mem_slots]
-            elif len(acquires) != len(mem_slots):
-                raise PulseError("#mem_slots must be equals to #acquires")
+        if mem_slots and len(acquire) != len(mem_slots):
```

Comment:
actually it will fail if we do not pass `mem_slots` or `mem_slot` :) so, I revert it

Reply:
if `mem_slots` is None then the condition is False, isn't it? Is that a failure? The behavior seems the same to me in both cases:
```
if A:
    if B:
        ...
```
should be the same as 
```
if A and B:
    ...
```
no?

Reply:
yes, you are right :) I was confised by reg_slots. In case of reg slots that trick would not work, due to it has else statement that set reg_slots to empty array

Reply:
I see! cool 👍